src/screens/DiscoveryScreenV2/index.js

```diff
@@ -223,7 +229,7 @@ const DiscoveryScreenV2 = ({route}) => {
     cancelTokenRef.current = axios.CancelToken.source();
   };
 
-  const RenderFragment = React.useMemo(() => {
+  const RenderFragment = (() => {
     if (selectedScreen === DISCOVERY_TAB_USERS)
       return (
         <UsersFragment
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [232-300]

The `RenderFragment` variable is now assigned using a function instead of `React.useMemo`. This change could potentially impact performance as `useMemo` ensures that expensive calculations are not run on every render, but only when certain dependencies have changed. If the calculation of `RenderFragment` is expensive and its value is used in the render method, consider reverting back to using `React.useMemo`.
```diff
- const RenderFragment = React.useMemo(() => {
+ const RenderFragment = (() => {
    if (selectedScreen === DISCOVERY_TAB_USERS)
      return (
        <UsersFragment
      );
    return null;
- }, [
-   selectedScreen,
-   searchText,
-   isLoadingDiscovery.topic,
-   isLoadingDiscovery.domain,
-   isLoadingDiscovery.news,
-   isLoadingDiscovery.user
- ]);
+ })();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->